crates/services/importer/src/ports.rs

```diff
@@ -32,21 +35,22 @@ pub trait Executor: Send + Sync {
 /// The database port used by the block importer.
 pub trait ImporterDatabase {
     /// Returns the latest block height.
-    fn latest_block_height(&self) -> StorageResult<BlockHeight>;
+    fn latest_block_height(&self) -> StorageResult<Option<BlockHeight>>;
     /// Update metadata about the total number of transactions on the chain.
     /// Returns the total count after the update.
     fn increase_tx_count(&self, new_txs_count: u64) -> StorageResult<u64>;
 }
 
 /// The port for returned database from the executor.
 pub trait ExecutorDatabase: ImporterDatabase {
-    /// Assigns the `Consensus` data to the block under the `block_id`.
-    /// Return the previous value at the `height`, if any.
-    fn seal_block(
+    /// Inserts the `SealedBlock` under the `block_id`.
+    // TODO: Remove `chain_id` from the signature, but for that transactions inside
+    //  the block should have `cached_id`. We need to guarantee that from the Rust-type system.
+    fn store_block(
         &mut self,
-        block_id: &BlockId,
-        consensus: &Consensus,
-    ) -> StorageResult<Option<Consensus>>;
+        chain_id: &ChainId,
+        block: &SealedBlock,
+    ) -> StorageResult<Option<()>>;
```

Comment:
Why not return `StorageResult<Bool>`?

Reply:
Also, could we have a self-documenting name? Why it's returning a value rather than `Result<()>` isn't self-evident. Maybe `store_block_if_new` or something like that.